distributed/client.py

```diff
@@ -1007,6 +1011,9 @@ def _ensure_connected(self, timeout=None):
         assert len(msg) == 1
         assert msg[0]["op"] == "stream-start"
 
+        if msg[0].get("warning"):
+            warnings.warn(msg[0]["warning"])
```

Comment:
Perhaps use a custom warning class instead of just `UserWarning` so users can filter on it if needed:

```python
class VersionMismatchWarning(Warning):
    """Indicates version mismatch between nodes"""
```

Reply:
Done